app/controllers/concerns/spree/admin/metadata_concern.rb

```diff
@@ -0,0 +1,19 @@
+module Spree
+  module Admin
+    module MetadataConcern
+      extend ActiveSupport::Concern
+
+      included do
+        def assert_metadata(object)
+          return unless params[:public_metadata].present?
+
+          params[:public_metadata][:key].each_with_index do |key, i|
+            next unless key.present?
```

Comment:
```suggestion
            next if key.blank?
```

Reply:
I was thinking about this logic last night, looking at how others set the keys using dot notation.

`sizing.european` -> ` 43`

Are other services using some logic to create nested json

```json
"sizing" {
    "european": "43"
}
```